doc/source/train/examples/intel_gaudi/sd.ipynb

```diff
@@ -0,0 +1,3473 @@
+    "\n",
+    "Therefore, it becomes straight-forward to make these changes:\n",
+    "```python\n",
+    "if __name__ == \"__main__\":\n",
+    "    import ray\n",
+    "    from ray import train\n",
+    "    from ray.train import ScalingConfig, Checkpoint, CheckpointConfig, RunConfig\n",
+    "    from ray.train.torch import TorchTrainer, TorchConfig\n",
+    "\n",
+    "    ray.init(address=\"auto\")\n",
+    "\n",
+    "    # Configure computation resources\n",
+    "    # In ScalingConfig, require an HPU for each worker\n",
+    "    scaling_config = ScalingConfig(num_workers=1, resources_per_worker={\"CPU\": 1, \"HPU\": 1})\n",
+    "    # Set backend to hccl in TorchConfig\n",
+    "    torch_config = TorchConfig(backend = \"hccl\")\n",
+    "    # Initialize a Ray TorchTrainer\n",
+    "    trainer = TorchTrainer(\n",
+    "        train_loop_per_worker=main,\n",
+    "        torch_config=torch_config,\n",
+    "        scaling_config=scaling_config,\n",
+    "    )\n",
+    "\n",
+    "    result = trainer.fit()\n",
+    "```\n",
+    "\n",
+    "Before we try to run, we need to inspect the `main` function to see if it can work in such way. As we skim through the function, it's clear that it does not take any input parameters, but it calls `parse_args` to get all configurations. Originally, these configurations are set in command line by MPI. But because we switch to Ray to start the workers, command line arguments are no longer accessible. Therefore, `parse_args` should be called in the main program and passed to `main` function.\n",
+    "\n",
+    "Apart from this, no other changes are necessary. By inserting the following code, you can now run the script on Ray."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "plaintext"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Replace lines below:\n",
+    "# def main():\n",
+    "#     args = parse_args()\n",
+    "# with these lines:\n",
+    "def main(config):\n",
+    "    args = config[\"args\"]"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "vscode": {
+     "languageId": "plaintext"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Replace lines below:\n",
+    "# if __name__ == \"__main__\":\n",
+    "#     main()\n",
+    "# with these lines:\n",
+    "if __name__ == \"__main__\":\n",
+    "    import ray\n",
+    "    from ray import train\n",
+    "    from ray.train import ScalingConfig, Checkpoint, CheckpointConfig, RunConfig\n",
+    "    from ray.train.torch import TorchTrainer, TorchConfig\n",
+    "\n",
+    "    ray.init(address=\"auto\")\n",
+    "\n",
+    "    # Configure computation resources\n",
+    "    # In ScalingConfig, require an HPU for each worker\n",
+    "    scaling_config = ScalingConfig(num_workers=1, resources_per_worker={\"CPU\": 1, \"HPU\": 1})\n",
```

Comment:
Hmm, I didn't want to make it too complicated here, and fine-tuning stable diffusion is not a task requiring >8 HPUs. But you are right, we should cover training across nodes somewhere.

Reply:
What about we mention something like "if you want to leverage multiple HPUs to accelerate training, you can do x,y,z changes." 

I saw in the text inversion example, it's using optimum `GaudiAccelerator`. I guess by default it will do DDP training if we have multiple HPUs? So the only change would be set `ScalingConfig(num_workers=)`?

Reply:
Correct. I mentioned this change(num_workers in ScalingConfig) in the document